src/Classes/WorkerTaxes/Payroll.php

```diff
@@ -144,28 +148,54 @@ public function getYtdEarnings(GovernmentalUnitArea $governmental_unit_area = nu
             return 0;
         }
 
-        return $this->wage_manager->calculateEarnings($area_income->getHistoricalWages(),
-            $this->start_date->copy()->startOfYear());
+        return $this->wage_manager->calculateEarnings(
+            $area_income->getHistoricalWages(),
+            $this->start_date->copy()->startOfYear()
+        );
+    }
+
+    public function getTipAmount(GovernmentalUnitArea $governmental_unit_area = null)
+    {
+        if ($governmental_unit_area === null) {
+            return 0;
+        }
+
+        /** @var AreaIncome $area_income */
+        $area_income = $this->area_incomes->get($governmental_unit_area->name);
+        if ($area_income === null) {
+            return 0;
+        }
+
+        return $this->wage_manager->calculateTipAmount($area_income->getWages()) / 100;
     }
 
     public function determineEarnings(TaxableIncome $taxable_income): void
     {
         $this->earnings = $this->wage_manager->calculateEarnings($taxable_income->getWages());
 
         $this->exempted_earnings = min($taxable_income->getExemptionAmountInCents() / 100, $this->earnings);
-        $this->exempted_supplemental_earnings = min(($taxable_income->getExemptionAmountInCents() / 100) - $this->exempted_earnings,
-            $this->supplemental_earnings);
+        $this->exempted_supplemental_earnings = min(
+            ($taxable_income->getExemptionAmountInCents() / 100) - $this->exempted_earnings,
+            $this->supplemental_earnings
+        );
 
         $start_of_year = $this->start_date->copy()->startOfYear();
-        $this->ytd_earnings = $this->wage_manager->calculateEarnings($taxable_income->getHistoricalWages(),
-            $start_of_year);
+        $this->ytd_earnings = $this->wage_manager->calculateEarnings(
+            $taxable_income->getHistoricalWages(),
+            $start_of_year
+        );
 
         $start_of_month = $this->start_date->copy()->startOfMonth();
-        $this->mtd_earnings = $this->wage_manager->calculateEarnings($taxable_income->getHistoricalWages(),
-            $start_of_month < $start_of_year ? $start_of_year : $start_of_month);
+        $this->mtd_earnings = $this->wage_manager->calculateEarnings(
+            $taxable_income->getHistoricalWages(),
+            $start_of_month < $start_of_year ? $start_of_year : $start_of_month
+        );
 
-        $this->supplemental_earnings = $this->wage_manager->calculateEarnings($taxable_income->getWages(),
-            null, true);
+        $this->supplemental_earnings = $this->wage_manager->calculateEarnings(
+            $taxable_income->getWages(),
+            null,
+            true
+        );
```

Comment:
set `tip_amount` here, so `$this->tip_amount = $this->wage_manager->calculateTipAmount($taxable_income->getWages())`